scripts/main/lychee_locale.js

```diff
@@ -238,6 +239,7 @@ lychee.locale = {
 		'ERROR_UNKNOWN'				: 'Something unexpected happened. Please try again and check your installation and server. Take a look at the readme for more information.',
 		'ERROR_LOGIN'				: 'Unable to save login. Please try again with another username and password!',
 		'ERROR_MAP_DEACTIVATED'				: 'Map functionality has been deactivated under settings.',
+		'ERROR_SEARCH_DEACTIVATED'				: 'Search functionality has been deactivated under settings.',
```

Comment:
I'm generally somewhat skeptical about adding elaborate notifications for cases that simply aren't supposed to happen. I'm just not sure if the amount of effort required to do it properly (adding them to all the language files on the server side) is worth it.

But it's your time and you choose how to use it :wink:.

Reply:
> For the reverse, I'd say that you should only be able to see tags if you have access to at least one photo with that tag.

@d7415 Could you explain in more detail what you mean? 
Let's assume we've got photo 1 (tag A), photo 2 (tag A & B) and photo 3 (no tag). All are pubic.
Do you mean:
(1) Tag A is only displayed at photos to user (in public mode), if at least one of the photos visible to him has tag A? (This would be imho trivial since you can't view tags without viewing a photo <- correct me if I'm wrong). 
(2) The section Tags is only shown, if at least one photo visible to the user (in public mode) has a tag -> This would require an additional db query for every photo and backend changes -> I'd vote against this because of (a) simplicity (what's the harm of showing an empty tag section) and (b) performance (I think I read somewhere that there has been recently an attempt to reduce the number of db queries to increase performance) 
(3) something completely different

Reply:
I've already spent the time and it has been merged :)